### DISCUSSION TITLE
Unable to start Dbeaver on MUSL

### DISCUSSION THREAD

**Discussion by cassiofb-dev** [dbeaver/dbeaver#19186] [DBeaver Development] [answered]
### Description

Hi, I downloaded and compiled Dbeaver on Void Linux (musl) and it was all fine.

<h2 align='center'>My Linux</h2>

![Void Linux](https://raw.githubusercontent.com/cassiofb-dev/void-linux-xfce4-themes-and-config/master/screenshot.png)

---

<h2 align='center'>Build Screenshot</h2>

![Build Screenshot](https://media.discordapp.net/attachments/1072625217282773003/1074214460211871805/image.png)

---

<h2 align='center'>What Happens</h2>

![Not Found](https://media.discordapp.net/attachments/1072625217282773003/1074443475426226206/image.png)

---

<h2 align='center'>Dependencies Info</h2>

![Dependencies Info](https://media.discordapp.net/attachments/1072625217282773003/1074216653333729321/image.png)

### DBeaver Version

devel

### Operating System

Void Linux (musl)

### Steps to reproduce

1. ```sh
    git clone https://github.com/dbeaver/dbeaver.git dbeaver
    cd dbeaver
    mvn package
    ```

### Additional context

_No response_

**Comment by E1izabeth** [comment]
Have you tried to run it without root privileges (without sudo)?

**Comment by cassiofb-dev** [comment]
Yep, I get the same result. I'm thinking the problem is something musl related, because dbeaver is available on void linux with glibc.

![image](https://user-images.githubusercontent.com/61878987/219252313-614b0b96-c2a2-42f0-ad79-52872a394136.png)

**Comment by Destrolaric** [comment]
@cassiofb-dev Did you try giving it execute permission through `chmod u+x dbeaver`?

**Comment by cassiofb-dev** [comment]
I got the same result. @Destrolaric is there another way to check dynamic dependencies besides ldd?

![image](https://user-images.githubusercontent.com/61878987/219903767-b88215fc-c5af-4c19-8ec3-9192b5ea6ee9.png)

**Comment by ShadelessFox** [comment]
Hello @cassiofb-dev,

What's the output of `file ./dbeaver`?

**Comment by cassiofb-dev** [comment]
Hi! By output you mean the contents? The file ``./debaver`` is a Linux binary.
![image](https://user-images.githubusercontent.com/61878987/221177598-cda9865e-94c7-4ca2-8d5b-a5556f6cc3bc.png)

**Comment by ShadelessFox** [comment]
Not the contents. This is a Linux program used for determining file type. https://linux.die.net/man/1/file

**Comment by cassiofb-dev** [comment]
![facepalm](https://media.tenor.com/BFVHX4oCxAAAAAAc/epic-facepalm.gif)

Sorry, I never had to use this command...

The output is:

```sh
[cassio@cassio dbeaver]$ file dbeaver
dbeaver: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=2f177540662891b66bfaff6ec2748390cbf9d123, with debug_info, not stripped, too many notes (256)
```

![image](https://user-images.githubusercontent.com/61878987/221294110-7e80ac51-6186-438e-b55a-247894b3142b.png)

**Comment by ShadelessFox** [comment, marked as answer]
Ah, I see. You can't run applications linked against glibc on Void Linux with musl.

You will have to compile the entire SWT and other native libraries that Eclipse uses. I can't give you any specific details, sorry.

You could give a try for this open-source utility: https://github.com/gch1p/voidnsrun

**Reply by cassiofb-dev** [reply]
Thanks, that's explained why it wasn't starting even when built on a musl system. I'll take some time to try the first approach building all tools on musl.
